src/types.ts

```diff
@@ -4329,6 +4329,38 @@ export interface ZodTemplateLiteralDef extends ZodTypeDef {
   typeName: ZodFirstPartyTypeKind.ZodTemplateLiteral;
 }
 
+export class ZodTemplateLiteralUnsupportedTypeError extends Error {
```

Comment:
Nice! I like having explicit errors for issues encountered on schema creation. With that said, I'm a little concerned if we're adding a lot of new functionality in a single PR. However, also we're having delays getting PRs reviewed so.. 🤷 

WDYT re: putting this in the `ZodError` file?

Reply:
Sounds good to me! 👍